Content.Server/GameObjects/Components/Construction/MachineBoardComponent.cs

```diff
@@ -14,32 +16,57 @@ namespace Content.Server.GameObjects.Components.Construction
     [RegisterComponent]
     public class MachineBoardComponent : Component, IExamine
     {
+        [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+
         public override string Name => "MachineBoard";
 
         [ViewVariables]
         private Dictionary<MachinePart, int> _requirements;
 
         [ViewVariables]
-        private Dictionary<StackType, int> _materialRequirements;
+        private Dictionary<string, int> _materialIdRequirements;
 
         [ViewVariables]
         private Dictionary<string, ComponentPartInfo> _componentRequirements;
 
         [ViewVariables(VVAccess.ReadWrite)]
         public string Prototype { get; private set; }
         public IReadOnlyDictionary<MachinePart, int> Requirements => _requirements;
-        public IReadOnlyDictionary<StackType, int> MaterialRequirements => _materialRequirements;
+        public IReadOnlyDictionary<string, int> MaterialIdRequirements => _materialIdRequirements;
+
+        public IEnumerable<KeyValuePair<StackPrototype, int>> MaterialRequirements
+        {
+            get
+            {
+                foreach (var (materialId, amount) in MaterialIdRequirements)
+                {
+                    var material = _prototypeManager.Index<StackPrototype>(materialId);
+                    yield return new KeyValuePair<StackPrototype, int>(material, amount);
+                }
+            }
+        }
+
         public IReadOnlyDictionary<string, ComponentPartInfo> ComponentRequirements => _componentRequirements;
 
         public override void ExposeData(ObjectSerializer serializer)
         {
             base.ExposeData(serializer);
             serializer.DataField(this, x => x.Prototype, "prototype", null);
             serializer.DataField(ref _requirements, "requirements", new Dictionary<MachinePart, int>());
-            serializer.DataField(ref _materialRequirements, "materialRequirements", new Dictionary<StackType, int>());
+            serializer.DataField(ref _materialIdRequirements, "materialRequirements", new Dictionary<string, int>());
             serializer.DataField(ref _componentRequirements, "componentRequirements", new Dictionary<string, ComponentPartInfo>());
         }
 
+        protected override void Startup()
+        {
+            base.Startup();
+
+            foreach (var material in _materialIdRequirements.Keys)
+            {
+                _prototypeManager.Index<StackPrototype>(material);
```

Comment:
Hmm, is this so it crashes if no such stack prototype exists?

Reply:
Yes. It validates the ids given so there are no typos, like TagComponent does.

Reply:
sounds good hten